mysql-test/main/subselect_mat_analyze_json.result

```diff
@@ -0,0 +1,948 @@
+set @save_optimizer_switch=@@optimizer_switch;
+create table t1 (a int);
+create table t2 (b int);
+insert into t1 values (null), (1), (2), (3);
+insert into t2 values (3), (4);
+set @@optimizer_switch = "materialization=on,in_to_exists=off,semijoin=off";
+explain format=json select * from t1 where a in (select b from t2);
+EXPLAIN
+{
+  "query_block": {
+    "select_id": 1,
+    "table": {
+      "table_name": "t1",
+      "access_type": "ALL",
+      "rows": 4,
+      "filtered": 100,
+      "attached_condition": "<in_optimizer>(t1.a,t1.a in (subquery#2))"
+    },
+    "subqueries": [
+      {
+        "materialization": {
+          "query_block": {
+            "select_id": 2,
+            "table": {
+              "table_name": "t2",
+              "access_type": "ALL",
+              "rows": 2,
+              "filtered": 100
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+# "Complete match" execution strategy
+analyze format=json select * from t1 where a in (select b from t2);
+ANALYZE
+{
+  "query_block": {
+    "select_id": 1,
+    "r_loops": 1,
+    "r_total_time_ms": "REPLACED",
+    "table": {
+      "table_name": "t1",
+      "access_type": "ALL",
+      "r_loops": 1,
+      "rows": 4,
+      "r_rows": 4,
+      "r_table_time_ms": "REPLACED",
+      "r_other_time_ms": "REPLACED",
+      "filtered": 100,
+      "r_filtered": 25,
+      "attached_condition": "<in_optimizer>(t1.a,t1.a in (subquery#2))"
+    },
+    "subqueries": [
+      {
+        "materialization": {
+          "r_strategy": "index_lookup",
+          "r_loops": 3,
+          "r_index_lookup_loops": 3,
+          "query_block": {
+            "select_id": 2,
+            "r_loops": 1,
+            "r_total_time_ms": "REPLACED",
+            "table": {
+              "table_name": "t2",
+              "access_type": "ALL",
+              "r_loops": 1,
+              "rows": 2,
+              "r_rows": 2,
+              "r_table_time_ms": "REPLACED",
+              "r_other_time_ms": "REPLACED",
+              "filtered": 100,
+              "r_filtered": 100
+            }
+          }
+        }
+      }
+    ]
+  }
+}
+# "Partial match" is used due to NOT IN
+# Force rowid-merge partial partial matching
+set @@optimizer_switch="partial_match_rowid_merge=on,partial_match_table_scan=off";
+analyze format=json select * from t1 where a not in (select b from t2);
+ANALYZE
+{
+  "query_block": {
+    "select_id": 1,
+    "r_loops": 1,
+    "r_total_time_ms": "REPLACED",
+    "table": {
+      "table_name": "t1",
+      "access_type": "ALL",
+      "r_loops": 1,
+      "rows": 4,
+      "r_rows": 4,
+      "r_table_time_ms": "REPLACED",
+      "r_other_time_ms": "REPLACED",
+      "filtered": 100,
+      "r_filtered": 50,
+      "attached_condition": "!<in_optimizer>(t1.a,t1.a in (subquery#2))"
+    },
+    "subqueries": [
+      {
+        "materialization": {
+          "r_strategy": "partial_match_merge",
```

Comment:
This looks confusing: it is not clear that partial matches are only done when we can't do index lookup.
How about changing to 

```index_lookup;array merge for partial match```

``` "r_partial_match_loops": 1,```
Please change to `r_partial_matches`.

``` "r_partial_merge_key_sizes": ["2"],```
please change to `r_partial_match_array_sizes` .

Reply:
Done